src/Mapping/Event/AdapterInterface.php

```diff
@@ -21,7 +21,9 @@
  *
  * @author Gediminas Morkevicius <gediminas.morkevicius@gmail.com>
  *
- * @method LifecycleEventArgs createLifecycleEventArgsInstance(object $object, ObjectManager $manager)
+ * @method LifecycleEventArgs createLifecycleEventArgsInstance(object $object, ObjectManager $manager) @deprecated Use createPreSoftDeleteEventArgs() or createPostSoftDeleteEventArgs() instead
+ * @method LifecycleEventArgs createPreSoftDeleteEventArgs(object $object, ObjectManager $manager)
+ * @method LifecycleEventArgs createPostSoftDeleteEventArgs(object $object, ObjectManager $manager)
```

Comment:
I would move these methods to [`SoftDeleteableAdapter`](https://github.com/doctrine-extensions/DoctrineExtensions/blob/058e95a34847b32d57b5583a006911a5465dac3f/src/SoftDeleteable/Mapping/Event/SoftDeleteableAdapter.php) since they are only related to that extension.